site/app/controllers/grading/ElectronicGraderController.php

```diff
@@ -47,7 +47,53 @@ private function validateDiffViewerType(string $type) {
         }
         return true;
     }
-
+/**
+        * Route for randomizing peer assignments
+        * @Route("/{_semester}/{_course}/gradeable/{gradeable_id}/ajaxRandomizePeers", methods={"POST"})
+        * @AccessControl(role="INSTRUCTOR")
+    */ 
+    public function ajaxRandomizePeers($gradeable_id){
+        $number_to_grade=1;
+       // $gradeable_id=$_POST['gradeable_id'];
+        $gradeable = $this->tryGetGradeable($gradeable_id);
+         if ($gradeable === false) {
+             $this->core->addErrorMessage('Invalid Gradeable!');
+             $this->core->redirect($this->core->buildCourseUrl());
```

Comment:
Because this is just an ajax call, we should be returning a failure here rather than redirecting.